src/Parallel/Invoke.hpp

```diff
@@ -17,7 +17,11 @@ namespace detail {
 inline bool max_inline_entry_methods_reached() noexcept {
   thread_local size_t approx_stack_depth = 0;
   approx_stack_depth++;
+#ifdef SPECTRE_PROFILING
+  if (approx_stack_depth < 0) {
```

Comment:
I see what you're getting at, but it seems silly to compare a size_t to 0. Consider `if(true) {`